ci-operator/step-registry/ibmcloud/provision/iamservice/minimal-permission/ibmcloud-provision-iamservice-minimal-permission-commands.sh

```diff
@@ -0,0 +1,78 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+#pre create the iam service-id and based it create the iam service api key for ibmcloud-ipi-minimal-permission test
+# based on doc https://docs.openshift.com/container-platform/4.14/installing/installing_ibm_cloud_public/installing-ibm-cloud-account.html#required-access-policies-ibm-cloud_installing-ibm-cloud-account
+
+function run_command() {
+    local CMD="$1"
+    echo "Running Command: ${CMD}"
+    eval "${CMD}"
+}
+
+# IBM Cloud CLI login
+function ibmcloud_login {
+    export IBMCLOUD_CLI=ibmcloud
+    export IBMCLOUD_HOME=/output
+    region="${LEASED_RESOURCE}"
+    export region
+    echo "Try to login..."
+    "${IBMCLOUD_CLI}" config --check-version=false
+    "${IBMCLOUD_CLI}" login -r ${region} --apikey @"${CLUSTER_PROFILE_DIR}/ibmcloud-api-key"
+}
+
+function createServicePolicy {
+  $IBMCLOUD_CLI iam service-policy-create "$1" --roles "$2" --service-name "$3"
+}
+#####################################
+##############Initialize#############
+#####################################
+ibmcloud_login
+
+CLUSTER_NAME="${NAMESPACE}-${UNIQUE_HASH}"
+serviceName="${CLUSTER_NAME}-perm"
+keyName="${serviceName}-key"
+keyFile=$(mktemp)
+
+## create the server id
+run_command "${IBMCLOUD_CLI} iam service-id-create ${serviceName}"
+echo ${serviceName} > "${SHARED_DIR}/iam_service_name"
+run_command "${IBMCLOUD_CLI} iam service-id ${serviceName}"
+
+
+## create the service key
+run_command "${IBMCLOUD_CLI} iam service-api-key-create ${keyName} ${serviceName} -d 'the key for minimal permission test in ${CLUSTER_NAME}' --file ${keyFile}"
```

Comment:
Do we need to remove the key in deprovision step?

Reply:
the key has been removed in cluster deprovision